### PR TITLE
[test-only] Api test. get applications list.

### PR DESCRIPTION
**Pull request by ScharfViktor**
- added tests for getting application list https://owncloud.dev/libre-graph-api/#/applications/ListApplications https://owncloud.testrail.io/index.php?/cases/view/2404

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiGraph-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/18853/44/1) failed. Further test are cancelled...

**Comment by ScharfViktor** [comment]
step
```
Then the user retrieve API response should contain the following applications information:
      | key                        | value                   |
      | displayName                | ownCloud Infinite Scale |
      | id                         | %uuid_v4%               |
      | appRoles@@@0@@@displayName | Admin                   |
      | appRoles@@@0@@@id          | %uuid_v4%               |
      | appRoles@@@1@@@displayName | User                    |
      | appRoles@@@1@@@id          | %uuid_v4%               |
      | appRoles@@@2@@@displayName | Space Admin             |
      | appRoles@@@2@@@id          | %uuid_v4%               |
      | appRoles@@@3@@@displayName | Guest                   |
      | appRoles@@@3@@@id          | %uuid_v4%               |
```

didn't work correct because we can get roles in random order so I split the step into two

File: tests/TestHelpers/GraphHelper.php [hunk 1]
```diff
@@ -499,43 +499,6 @@ public static function deleteGroup(
...
-	 * @param string $adminUser
-	 * @param string $adminPassword
-	 * @param string $groupId
-	 * @param array $users expects users array with user ids
-	 *                     [ [ 'id' => 'some_id' ], ]
-	 *
-	 * @return ResponseInterface
-	 * @throws GuzzleException
-	 */
-	public static function addUsersToGroup(
```
Comment (individual-it) [inline tests/TestHelpers/GraphHelper.php:514]: why deleting this function?

File: tests/acceptance/features/bootstrap/GraphContext.php [hunk 2]
```diff
@@ -1459,4 +1459,62 @@ public function theResponseShouldContainTheFollowingDriveInformation(TableNode $
...
+	 * @Then the user retrieve API response should contain the following applications information:
+	 *
+	 * @return void
+	 */
+	public function theResponseShouldContainTheFollowingApplicationsInformation(TableNode $table): void {
+		Assert::assertIsArray($responseArray = ($this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse()))['value'][0]);
+		foreach ($table->getHash() as $row) {
+			$key = $row["key"];
+			if ($key === 'id') {
+				Assert::assertTrue(GraphHelper::isUUIDv4($responseArray[$key]), __METHOD__ . ' Expected id to have UUIDv4 pattern but found: ' . $row["value"]);
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/GraphContext.php:1493]:
```suggestion
				Assert::assertTrue(
				    GraphHelper::isUUIDv4($responseArray[$key]),
				    __METHOD__ . ' Expected id to have UUIDv4 pattern but found: ' . $row["value"]
				);
```
just to make the review easier

**Review by individual-it — CHANGES_REQUESTED** [review]
why did the go files change?

File: tests/acceptance/features/bootstrap/GraphContext.php [hunk 3]
```diff
@@ -1459,4 +1459,62 @@ public function theResponseShouldContainTheFollowingDriveInformation(TableNode $
...
+	 * @return void
+	 */
+	public function theResponseShouldContainTheFollowingAppRolesInformation(TableNode $table): void {
+		Assert::assertIsArray($responseArray = ($this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse()))['value'][0]);
+		
+		foreach ($table->getRows() as $row) {
+			$foundRoleInResponse = false;
+			foreach ($responseArray['appRoles'] as $role) {
+				if ($role['displayName'] === $row[0]) {
+					$foundRoleInResponse = true;
```
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/GraphContext.php:1514]:
```suggestion
					$foundRoleInResponse = true;
					break;
```
can we add a break to this logic?

File: tests/TestHelpers/GraphHelper.php (hunk 1 above)
Comment (ScharfViktor) [reply tests/TestHelpers/GraphHelper.php:514]: - We don't have a single step that uses this method.
- we have no such endpoint in the list and the method does not work

**Comment by ScharfViktor** [comment]
> why did the go files change?

my mistake. I'll bring them back.

**Review by butonic — APPROVED** [review]

**Review by SagarGi — COMMENTED** [review]
lgtm :+1:

**Review by individual-it — APPROVED** [review]

**Pull request merged**